### PR TITLE
Editor: Limit spotlight mode to the editor.

### LINKED ISSUE #42660: Patterns: Spotlight Mode Option makes patterns in inserter look greyed out
### Description

This appears to be a defect, as I cannot think of any reason why enabling Spotlight mode should affect the Block Inserter sidebar interface. 

Below is the information I originally posted on the WordPress.org support forum. The response I got there confirmed the behavior.

Using the Gutenberg editor with Spotlight mode turned on is the way I like to work, as many times I cannot otherwise quickly determine which block is selected.

Recently, I was viewing the Patterns tab of the Block Inserter sidebar, and it was displaying all of the available patterns grayed out, or ghosted. This made it difficult to determine what the pattern would render as on a page, and of course, clicking a pattern adds it to the page, when all I really want at that moment is to find a pattern to use. Once I know which pattern I want, then I will add it to the page.

Anyhow, I wondered why the patterns were all ghosted, and it turns out the culprit is Spotlight Mode. If that is turned off, the Patterns tab in the Block Inserter shows the available patterns in a normal, and in my view, useful way. Note that this ghosting also happens with the Block/Pattern/Reusable Block preview shown when hovering over an item on one of those tabs. Overall, this ghosting behavior is less than useful in my view.

Granted, any user could just toggle the Spotlight Mode option on/off as needed, but that is awkward, and I really think that Spotlight Mode should only affect the page or template content that is being worked on, not the sidebar.

Is this considered a feature, or a bug?
Is there a way to change this behavior without having to changing the options each time I use the Pattern tab?
Is there something I am missing here?

Thanks!

### Step-by-step reproduction instructions

1. Enable Spotlight Mode Option
2. Open Block Inserter sidebar
3. Click the Patterns tab
4. Note that the available patterns all appear grayed/ghosted, making it difficult to visualize the available patterns
5. Disable Spotlight Mode Option
6. Again view the Patterns tab
7. Note that the available patterns are not grayed/ghosted. This is how I feel it should always be, even if Spotlight Mode is enabled
8. I attempted to add a screen capture, but keep getting a "Something went really wrong, and we can't process the file." message, but seeing that this was confirmed by [threadi](https://wordpress.org/support/users/threadi/) (@threadi) on the WordPress Forum, I don't think it is totally necessary at this time

### Screenshots, screen recording, code snippet

_No response_

### Environment info

- WordPress 6.0.1
- Edge Version 103.0.1264.62 (Official build) (64-bit)
- Windows 11 Pro

### Please confirm that you have searched existing issues in the repo.

Yes

### Please confirm that you have tested with all plugins deactivated except Gutenberg.

Yes

### PR DESCRIPTION
**Pull request by juanfra**
## What?

Enabling spotlight mode in the editor greyed out every block representation outside the editor (preview in the inserter, patterns in the inserter, etc).

Setting "Spotlight mode" to false when calling the block previewer.

Fixes https://github.com/WordPress/gutenberg/issues/42660

## Why?

Spotlight mode should be limited to the editor content. When affecting the inserter and other previews, it makes it look like a defect.

## How?

Setting "Spotlight mode" to false when calling the block previewer.

## Testing Instructions

**Post editor**

1. Open a post or page.
2. Enable Spotlight Mode Option.
3. Open the block inserter sidebar.
4. Confirm that the blocks preview and patterns previews are not greyed.
5. Add content to the post, use any block and confirm that spotlight mode is still working as expected.

**Site editor**

1. Open the site editor. (Assuming that Spotlight Mode is ON)
2. Click on the preview to enter edit mode.
3. Open the block inserter sidebar.
4. Confirm that the blocks preview and patterns previews are not greyed.
5. Click on any element of the page you're editing, and confirm that spotlight mode is still working.
6. Open the stylebook, confirm that the stylebook is not greyed and when you click on any element, the preview is also not greyed (bonus fix with this).

## Screenshots or screencast 

https://github.com/WordPress/gutenberg/assets/252415/d472623c-0d76-4364-a519-66c2fcf510b0

### CODE REVIEW

**Comment by Mamaduka** [comment]
Hi, @juanfra,

I think we can also disable "Spotlight mode" mode via the block editor settings for previews.

```diff
diff --git packages/block-editor/src/components/block-preview/index.js packages/block-editor/src/components/block-preview/index.js
index 0fb7f55b995..245d0ee4348 100644
--- packages/block-editor/src/components/block-preview/index.js
+++ packages/block-editor/src/components/block-preview/index.js
@@ -54,7 +54,11 @@ export function BlockPreview( {
 		[]
 	);
 	const settings = useMemo(
-		() => ( { ...originalSettings, __unstableIsPreviewMode: true } ),
+		() => ( {
+			...originalSettings,
+			focusMode: false, // Disable "Spotlight mode".
+			__unstableIsPreviewMode: true,
+		} ),
 		[ originalSettings ]
 	);
 	const renderedBlocks = useMemo(
@@ -117,6 +121,7 @@ export function useBlockPreview( { blocks, props = {}, layout } ) {
 		() => ( {
 			...originalSettings,
 			styles: undefined, // Clear styles included by the parent settings, as they are already output by the parent's EditorStyles.
+			focusMode: false, // Disable "Spotlight mode".
 			__unstableIsPreviewMode: true,
 		} ),
 		[ originalSettings ]

```

**Comment by juanfra** [comment]
Thanks @Mamaduka  ✨

That's indeed a way better approach, I've updated the PR

File: packages/edit-site/src/components/style-book/index.js [hunk 1]
```diff
@@ -222,7 +222,11 @@ function StyleBook( {
 	);
 
 	const settings = useMemo(
-		() => ( { ...originalSettings, __unstableIsPreviewMode: true } ),
+		() => ( {
+			...originalSettings,
+			focusMode: false, // Disable "Spotlight mode".
+			__unstableIsPreviewMode: true,
+		} ),
```
Comment (Mamaduka) [inline packages/edit-site/src/components/style-book/index.js]: I think this change in the `StyleBook` component isn't required; these `settings` aren't passed into `BlockEditorProvider`.

Side note: The component doesn't really need the complete set of settings. They're passed to `StyleBookBody`, which only uses `settings.styles`. But this is unrelated to the fix here.

cc @andrewserong, @ramonjd

**Review by Mamaduka — APPROVED** [review]
Thank you, @juanfra! The fix works as expected!

**Comment by juanfra** [comment]
Thank you, George!

Oh snap, I just realized there's a new way to credit the changes by adding the people involved in the merge commit, and I didn't do that. Sorry 🙏

**Comment by getdave** [comment]
@juanfra As this fixed a bug on the WP 6.5 board I have adde the appropriate label to ensure it gets backported. If that is _not_ correct please let me know.

**Comment by getdave** [comment]
I just _manually_ cherry-picked this PR to the [cherry-pick-wp-6-5-beta-3 branch](https://github.com/WordPress/gutenberg/pull/59394) to get it included in the next release.

**Comment by youknowriad** [comment]
Instead of explicitly passing this setting, I wonder if `__unstableIsPreviewMode` should do that automatically. In other words, only allow enabling spotlight mode if we're not in preview mode.

**Pull request merged**